### ISSUE TITLE
Add paket into investigation?

### ISSUE THREAD

**Issue by fffej** [red-gate/internal-tech-radar#21]
https://fsprojects.github.io/Paket/

I think @alastairs has some experience with this? Is this a technology we should consider adopting more widely?

**Comment by alastairs** [comment]
Yes, we used it on the SSDT project. I think @nicolasvan is also using it for some build dependencies. 

It solves a lot of issues with NuGet, but the Visual Studio integration is not nearly as good (it's not an official thing from either Paket or Microsoft). This caused us some issues in SSDT (I forget the details), but using it from the command-line works completely fine. 

There's a handy convert-from-NuGet option on the command-line too.

**Comment by nicolasvan** [comment]
I think it's probably worth investigating more. It does work really well on the command line, so quite good for our build scripts. (I would be inclined to use it more as a replacements for calls to `nuget.exe install`).

How it works from within VS and how it impact our .csproj might be more controversial though... (_I rarely use VS so can't really comment on that_)

Could the SSDT team give more details about issues they encountered? Would they use paket again? Or was it just to painful?
Maybe @ChrisHurley, @DavidSimner, @smobs, @alextercete have some opinion. (just took you guys from our list of contributors to our SSDT repos) :smile:

**Comment by ChrisHurley** [comment]
The main things that come to mind about it are:
* It doesn't support running any scripts when adding/installing packages. On many levels this is no bad thing, but it might cause problems with some packages.
* When we used it on the SSDT team, the Visual Studio integration was somewhat painful, as the search kept breaking, it was easy for the window to get stuck open in the background, and it would somehow manage to lose unsaved changes to unrelated files as well as the `.csproj` itself. I ended up just using PowerShell. I don't know if it's improved yet.
* It makes a lot of noise in the `.csproj` files to support different platforms/framework versions; this is probably more correct but adding a new package can result in many lines changed.

However these problems aren't insurmountable.
